xonsh/built_ins.py

```diff
@@ -535,6 +535,9 @@ def prep_preexec_fn(self, kwargs, pipeline_group=None):
         """Prepares the 'preexec_fn' keyword argument"""
         if not (ON_POSIX and self.cls is subprocess.Popen):
             return
+        env = builtins.__xonsh_env__
+        if not env['XONSH_INTERACTIVE']:
```

Comment:
I think this is safe. this env is always set during the start up phase, and here is a run-sub-proc phase. But I could update a bit :)